packages/signer/src/authClient.ts

```diff
@@ -44,13 +69,141 @@ type PendingOtp = {
   orgId: string;
 };
 
+/**
+ * Reads and removes the specified query params from the URL.
+ *
+ * @param {T} keys object whose values are the query parameter keys to read and
+ * remove
+ * @returns {{ [K in keyof T]: string | undefined }} object with the same keys
+ * as the input whose values are the values of the query params.
+ */
+function getAndRemoveQueryParams<T extends Record<string, string>>(
```

Comment:
This function, and the functions that call it, will fail if not in a browser and so probably belong in the `signer-web` package.